src/index/VocabularyMergerImpl.h

```diff
@@ -63,25 +65,38 @@ auto VocabularyMerger::mergeVocabulary(const std::string& basename,
     return lessThan(p1.entry_, p2.entry_);
   };
 
-  std::vector<cppcoro::generator<QueueWord>> generators;
+  struct QueueWordGenerator : ad_utility::InputRangeFromGet<QueueWord> {
+    QueueWordGenerator(const size_t fileIdx, const std::string& basename)
+        : infile{absl::StrCat(basename, PARTIAL_VOCAB_FILE_NAME, fileIdx)},
+          fileIdx{fileIdx} {
+      infile >> numWords;
+    }
+
+    std::optional<QueueWord> get() {
+      if (i < numWords) {
+        infile >> val;
+        ++i;
+        QueueWord word{std::move(val), fileIdx};
+        return word;
+      }
 
-  auto makeGenerator = [&](size_t fileIdx) -> cppcoro::generator<QueueWord> {
-    ad_utility::serialization::FileReadSerializer infile{
-        absl::StrCat(basename, PARTIAL_VOCAB_FILE_NAME, fileIdx)};
+      return std::nullopt;
+    }
+
+    ad_utility::serialization::FileReadSerializer infile;
     uint64_t numWords;
-    infile >> numWords;
+    uint64_t i{0};
+    const std::size_t fileIdx;
     TripleComponentWithIndex val;
-    for ([[maybe_unused]] auto idx : ad_utility::integerRange(numWords)) {
-      infile >> val;
-      QueueWord word{std::move(val), fileIdx};
-      co_yield word;
-    }
   };
 
+  std::vector<ad_utility::InputRangeTypeErased<QueueWord>> generators;
+
   // Open and prepare all infiles and mmap output vectors.
   generators.reserve(numFiles);
   for (size_t i = 0; i < numFiles; i++) {
-    generators.push_back(makeGenerator(i));
+    generators.push_back(ad_utility::InputRangeTypeErased<QueueWord>{
+        QueueWordGenerator{i, basename}});
```

Comment:
emplace_back should allow you to drop the lengthy `InputRangeTypErased...` type.